frontend/src/pages/projects/screens/detail/notebooks/useNotebookImageData.ts

```diff
@@ -1,52 +1,92 @@
 import * as React from 'react';
-import { ImageStreamKind, ImageStreamSpecTagType, NotebookKind } from '~/k8sTypes';
+import { NotebookKind } from '~/k8sTypes';
 import useNamespaces from '~/pages/notebookController/useNamespaces';
 import useImageStreams from '~/pages/projects/screens/spawner/useImageStreams';
 import { NotebookContainer } from '~/types';
+import { getImageStreamDisplayName } from '~/pages/projects/screens/spawner/spawnerUtils';
+import { NotebookImageAvailability } from './const';
+import { NotebookImageData } from './types';
 
-const useNotebookImageData = (
-  notebook?: NotebookKind,
-): [
-  data: { imageStream: ImageStreamKind; imageVersion: ImageStreamSpecTagType } | null,
-  loaded: boolean,
-] => {
+const useNotebookImageData = (notebook?: NotebookKind): NotebookImageData => {
   const { dashboardNamespace } = useNamespaces();
-  const [images, loaded] = useImageStreams(dashboardNamespace);
+  const [images, loaded, loadError] = useImageStreams(dashboardNamespace, true);
 
   return React.useMemo(() => {
-    if (!notebook) {
-      return [null, false];
+    if (!notebook || !loaded) {
+      return [null, false, undefined];
     }
 
-    if (!loaded) {
-      return [null, false];
+    if (loadError) {
+      return [null, true, loadError];
```

Comment:
I don't think loaded should be set to `true` here because this pattern would suggests that data is available.
```suggestion
      return [null, false, loadError];
```

Reply:
See comment above